tools/codegen/codespec/model.go

```diff
@@ -21,69 +21,69 @@ type Model struct {
+	Description        *string `yaml:"description,omitempty"`
+	DeprecationMessage *string `yaml:"deprecation_message,omitempty"`
 
-	Attributes Attributes
+	Attributes Attributes `yaml:"attributes"`
 }
 
 type Attributes []Attribute
 
 // Add this field to the Attribute struct
 // Usage AttributeUsage
 type Attribute struct {
-	Set                      *SetAttribute
-	String                   *StringAttribute
-	Float64                  *Float64Attribute
-	List                     *ListAttribute
-	Bool                     *BoolAttribute
-	ListNested               *ListNestedAttribute
-	Map                      *MapAttribute
-	MapNested                *MapNestedAttribute
-	Number                   *NumberAttribute
-	Int64                    *Int64Attribute
-	Timeouts                 *TimeoutsAttribute
-	SingleNested             *SingleNestedAttribute
-	SetNested                *SetNestedAttribute
-	Description              *string
-	DeprecationMessage       *string
-	CustomType               *CustomType
-	ComputedOptionalRequired ComputedOptionalRequired
-	TFSchemaName             stringcase.SnakeCaseString
-	TFModelName              string
-	ReqBodyUsage             AttributeReqBodyUsage
-	Sensitive                bool
-	CreateOnly               bool // leveraged for defining plan modifier which avoids updates on this attribute
+	Set                      *SetAttribute              `yaml:"set,omitempty"`
+	String                   *StringAttribute           `yaml:"string,omitempty"`
+	Float64                  *Float64Attribute          `yaml:"float64,omitempty"`
+	List                     *ListAttribute             `yaml:"list,omitempty"`
+	Bool                     *BoolAttribute             `yaml:"bool,omitempty"`
+	ListNested               *ListNestedAttribute       `yaml:"list_nested,omitempty"`
+	Map                      *MapAttribute              `yaml:"map,omitempty"`
+	MapNested                *MapNestedAttribute        `yaml:"map_nested,omitempty"`
+	Number                   *NumberAttribute           `yaml:"number,omitempty"`
+	Int64                    *Int64Attribute            `yaml:"int64,omitempty"`
+	Timeouts                 *TimeoutsAttribute         `yaml:"timeouts,omitempty"`
+	SingleNested             *SingleNestedAttribute     `yaml:"single_nested,omitempty"`
+	SetNested                *SetNestedAttribute        `yaml:"set_nested,omitempty"`
+	Description              *string                    `yaml:"description,omitempty"`
+	DeprecationMessage       *string                    `yaml:"deprecation_message,omitempty"`
+	CustomType               *CustomType                `yaml:"custom_type,omitempty"`
+	ComputedOptionalRequired ComputedOptionalRequired   `yaml:"computed_optional_required"`
+	TFSchemaName             stringcase.SnakeCaseString `yaml:"tf_schema_name"`
+	TFModelName              string                     `yaml:"tf_model_name"`
+	ReqBodyUsage             AttributeReqBodyUsage      `yaml:"req_body_usage"`
```

Comment:
We could consider changing the type of ReqBodyUsage to string similar to how we handle ComputedOptionalRequired, makes it readable in the model.